src/video/vita/SDL_vitamessagebox.c

```diff
@@ -0,0 +1,153 @@
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "../../SDL_internal.h"
+
+#if SDL_VIDEO_DRIVER_VITA
+
+#include "SDL_vitavideo.h"
+#include "SDL_vitamessagebox.h"
+#include <psp2/message_dialog.h>
+
+#if SDL_VIDEO_RENDER_VITA_GXM
+#include "../../render/vitagxm/SDL_render_vita_gxm_tools.h"
+#endif /* SDL_VIDEO_RENDER_VITA_GXM */
+
+int VITA_ShowMessageBox(const SDL_MessageBoxData *messageboxdata, int *buttonid)
+{
+    SceCommonDialogConfigParam commonDialogConfigParam;
+    SceMsgDialogParam param;
+    SceMsgDialogUserMessageParam msgParam;
+    SceMsgDialogButtonsParam buttonParam;
+    SceDisplayFrameBuf dispparam;
+    
+    SceCommonDialogUpdateParam updateParam;
+    SceMsgDialogResult result;
+    bool setup_minimal_gxm = false;
+#if !SDL_VIDEO_RENDER_VITA_GXM
+    {
+        return -1;
+    }
+#endif
+#if SDL_VIDEO_RENDER_VITA_GXM
+    if (messageboxdata->numbuttons > 3)
+    {
+        return -1;
+    }
+    SDL_zero(commonDialogConfigParam);
+    sceCommonDialogSetConfigParam(&commonDialogConfigParam);
+    sceMsgDialogParamInit(&param);
+    param.mode = SCE_MSG_DIALOG_MODE_USER_MSG;
+    SDL_zero(msgParam);
+    msgParam.msg = (const SceChar8*)messageboxdata->title;
+    SDL_zero(buttonParam);
+    if (messageboxdata->numbuttons == 3)
+    {
+        msgParam.buttonType = SCE_MSG_DIALOG_BUTTON_TYPE_3BUTTONS;
+        msgParam.buttonParam = &buttonParam;
+        buttonParam.msg1 = messageboxdata->buttons[0].text;
+        buttonParam.msg2 = messageboxdata->buttons[1].text;
+        buttonParam.msg3 = messageboxdata->buttons[2].text;
+    }
+    else if (messageboxdata->numbuttons == 2)
+    {
+        msgParam.buttonType = SCE_MSG_DIALOG_BUTTON_TYPE_YESNO;
+    }
+    else if (messageboxdata->numbuttons == 1)
+    {
+        msgParam.buttonType = SCE_MSG_DIALOG_BUTTON_TYPE_OK;
+    }
+    param.userMsgParam = &msgParam;
+
+    dispparam.size = sizeof(dispparam);
+
+    // Setup display if it hasn't been initialized before
+    if (sceDisplayGetFrameBuf(&dispparam, SCE_DISPLAY_SETBUF_NEXTFRAME) < 0)
+    {
+        gxm_minimal_init_for_common_dialog();
+        setup_minimal_gxm = true;
+    }
+
+    gxm_init_for_common_dialog();
+    
+    if (sceMsgDialogInit(&param) >= 0 && sceDisplayGetFrameBuf(&dispparam, SCE_DISPLAY_SETBUF_NEXTFRAME) >= 0)
+    {
+        while (sceMsgDialogGetStatus() == SCE_COMMON_DIALOG_STATUS_RUNNING)
+        {
+            SDL_zero(updateParam);
+
+            updateParam.renderTarget.colorFormat    = dispparam.pixelformat;
+            updateParam.renderTarget.surfaceType    = SCE_GXM_COLOR_SURFACE_LINEAR;
+            updateParam.renderTarget.width          = dispparam.width;
+            updateParam.renderTarget.height         = dispparam.height;
+            updateParam.renderTarget.strideInPixels = dispparam.pitch;
+
+            updateParam.renderTarget.colorSurfaceData = dispparam.base;
```

Comment:
Shouldn't that be `buffer_for_common_dialog[back_buffer_index_for_common_dialog]`?